backend/src/main/java/com/jobwait/stat/impl/JobTitleStat.java

```diff
@@ -0,0 +1,41 @@
+package com.jobwait.stat.impl;
+
+import java.util.HashMap;
+import java.util.List;
+
+import com.jobwait.domain.Answer;
+import com.jobwait.domain.Stat;
+import com.jobwait.stat.AnswerFetching;
+
+public class JobTitleStat extends Stat implements AnswerFetching {
+    public JobTitleStat(String id) {
+        super(id);
+        this.rows = new HashMap<String, Object>();
+    }
+
+    // todo - put this into a defaultdict util class
+    private void addOrIncrementRow(String title) {
+        if (this.rows.containsKey(title)) {
+            int prev = (int) this.rows.get(title);
+            this.rows.put(title, prev + 1);
+        } else {
+            this.rows.put(title, 1);
+        }
+    }
+
+    public void fetchAndSetRows() {
+        this.rows = new HashMap<String, Object>();
+
+        List<List<Answer>> answers = this.fetchAllAnswerSets();
+        for (List<Answer> answerSet : answers) {
+            for (Answer answer : answerSet) {
+                if (answer.questionKey.equals("jobtitle")) {
+                    String title = answer.answerValueAsString();
+                    if (title != null) {
+                        this.addOrIncrementRow(answer.answerValueAsString());
+                    }
+                }
+            }
```

Comment:
if my previous question about Optional makes no sense (because I misunderstood why we have List of List) disregard this: if we had optional here we wouldnt have to do the double for loop

Reply:
I think you might have misunderstood the shape of the data. I responded to your other comment. I'll mark this as resolved (or you can too) if the question no longer applies.